src/tiled/tilesetdock.cpp

```diff
@@ -758,20 +758,31 @@ void TilesetDock::exportTileset()
     if (!tileset)
         return;
 
+    Preferences *prefs = Preferences::instance();
+
     const QLatin1String extension(".tsx");
-    QString suggestedFileName = QFileInfo(mMapDocument->fileName()).path();
-    suggestedFileName += QLatin1Char('/');
+    QLatin1String empty(".");
+    QString suggestedFileName = prefs->lastPath(Preferences::ExternalTileset);
+
+    for(int i = (suggestedFileName.length() - 1); suggestedFileName.at(i) != QLatin1Char('/')
+        && suggestedFileName != empty; i--)
+        suggestedFileName.remove(i, 1);
+
+    if(suggestedFileName == empty)
+        suggestedFileName += QLatin1Char('/');
     suggestedFileName += tileset->name();
-    if (!suggestedFileName.endsWith(extension))
```

Comment:
Why did you remove this condition? I think it should stay to avoid adding the extension twice when it is part of the tileset name.